docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+
+[Implementation:]{.ul}
+
+1.  A command is requested from the user as the application starts,
+    > Duke.run() calls UI.scanInput() to read the user's input from the
+    > command line.
+
+2.  The user input determines which type of functionalities will be
+    > accessed by the user.
+
+**Staff Menu**
+
+[Implementation:]{.ul}
+
+1.  When "Staff" is given as input in the main menu, duke.run() will
+    > call StaffParser.run()
+
+2.  StaffParser.run() provides the user a huge range of functionalities
+    > to work with the Staff Objects.
+
+3.  StaffParser.run() will request input from the user.
+
+4.  User input will determine the type of actions taken on Staff
+    > Objects.
+
+5.  The details of the functionalities related to Staff Object is
+    > detailed in the section below under [[Staff-related
+    > Features]{.ul}](#staff-related-features).
+
+**Staff Menu**
+
+This feature allows users to select the different menus for Patient /
+Staff / Doctor Appointments / Nurse Schedules and Drugs through
+inputting commands within the main menu. Failure to input a correct or
+recognized command will then prompt the program to generate an exception
+that will alert the user of what they have done wrongly, and also prompt
+the user to access the help list to view the correct set of commands and
+their proper syntaxes.
+
+If the user enters a correct and recognized command, they will be able
+to access the functionalities related to the specific Objects (i.e
+Staff, Patient, Doctor Appointments, Nurse Schedules, Drugs).
+
+**Doctor Appointment Menu**
+
+This feature of the program can be accessed through the main menu as the
+application starts running. Accessing this feature gains access to the
+DoctorAppointment package, including the classes
+AppointmentActions.java, DoctorAppointment.java,
+DoctorAppointmentInstance.java, DoctorAppointmentStorage.java and
+Parser.java. The purpose of this feature is to save a database of
+patient details when they make an appointment with the doctor.
+
+[Implementation:]{.ul}
+
+6.  When the user types in a command,
+
+    a.  Duke.run() calls UI.scanInput() to read the user's input from
+        > the command line.
+
+7.  After scanning the user input, it will be run through case
+    > statements in Duke.run() to find the appropriate command that
+    > corresponds to the user's input
+
+8.  Depending on the input Duke.run() will then bring the user to the
+    > specific instance handler for that feature, which can be
+    > Staff.Parser.run() or NurseScheduleInstance.main(). Alternatively,
+    > Duke.run() creates either a PatientCommandInstance or
+    > DoctorAppointmentInstance or DrugInstance object that may act as
+    > an instance handler.
+
+9.  The selected instance handler will then have its own menu that may
+    > handle any additional user commands that are inputted to access
+    > features within that particular instance.
+
+### **Staff-related Features**
```

Comment:
I think visual representations of the sequence of actions can be included to accompany the description.